alert-common/src/main/java/com/synopsys/integration/alert/common/channel/MessageSplitter.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.channel;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+//TODO use this for SlackChannel to create message chunks.
+public final class MessageSplitter {
+    public static final String DEFAULT_LINE_SEPARATOR = "\n";
+    private final int limit;
+    private final String lineSeparator;
+
+    public MessageSplitter(final int limit) {
+        this(limit, DEFAULT_LINE_SEPARATOR);
+    }
+
+    public MessageSplitter(final int limit, final String lineSeparator) {
+        this.limit = limit;
+        this.lineSeparator = lineSeparator;
+    }
+
+    public final int getLimit() {
+        return limit;
+    }
+
+    public final String getLineSeparator() {
+        return lineSeparator;
+    }
+
+    public final List<String> splitMessages(final Collection<String> messagePieces) {
+        return splitMessages(messagePieces, false);
+    }
+
+    public final List<String> splitMessages(final Collection<String> messagePieces, boolean separateChunksByLineSeparator) {
```

Comment:
Do you think there's a library that might already do a lot of these things? If so, it might be easier to just use the library instead of creating our own chunker.

Reply:
There might be but this is a bug fix.  I want to use the code we have than introduce a new dependency at this point.   Maybe for the next release.